packages/quick_actions/quick_actions_android/test/quick_actions_android_test.dart

```diff
@@ -4,169 +4,100 @@
-
-      test('clearShortcutItems', () {
-        final QuickActionsAndroid quickActions = buildQuickActionsPlugin();
-        quickActions.clearShortcutItems();
-        expect(
-          log,
-          <Matcher>[
-            isMethodCall('clearShortcutItems', arguments: null),
-          ],
-        );
-        log.clear();
-      });
-    });
+  test('setShortCutItems', () async {
+    await quickActions.initialize((String type) {});
+    const ShortcutItem item =
+        ShortcutItem(type: 'test', localizedTitle: 'title', icon: 'icon.svg');
+    await quickActions.setShortcutItems(<ShortcutItem>[item]);
+
+    expect(api.items.first.type, item.type);
+    expect(api.items.first.localizedTitle, item.localizedTitle);
+    expect(api.items.first.icon, item.icon);
   });
 
-  group('$ShortcutItem', () {
-    test('Shortcut item can be constructed', () {
-      const String type = 'type';
-      const String localizedTitle = 'title';
-      const String icon = 'foo';
+  test('clearShortCutItems', () {
+    quickActions.initialize((String type) {});
+    const ShortcutItem item =
+        ShortcutItem(type: 'test', localizedTitle: 'title', icon: 'icon.svg');
+    quickActions.setShortcutItems(<ShortcutItem>[item]);
+    quickActions.clearShortcutItems();
+
+    expect(api.items.isEmpty, true);
+  });
 
-      const ShortcutItem item =
-          ShortcutItem(type: type, localizedTitle: localizedTitle, icon: icon);
+  test('Shortcut item can be constructed', () {
+    const String type = 'type';
+    const String localizedTitle = 'title';
+    const String icon = 'foo';
 
-      expect(item.type, type);
-      expect(item.localizedTitle, localizedTitle);
-      expect(item.icon, icon);
-    });
+    const ShortcutItem item =
+        ShortcutItem(type: type, localizedTitle: localizedTitle, icon: icon);
+
+    expect(item.type, type);
+    expect(item.localizedTitle, localizedTitle);
+    expect(item.icon, icon);
   });
 }
 
-/// This allows a value of type T or T? to be treated as a value of type T?.
-///
-/// We use this so that APIs that have become non-nullable can still be used
-/// with `!` and `?` on the stable branch.
-T? _ambiguate<T>(T? value) => value;
+class _FakeQuickActionsApi implements AndroidQuickActionsApi {
+  List<ShortcutItem> items = <ShortcutItem>[];
+  bool getLaunchActionCalled = false;
+
+  @override
+  Future<void> clearShortcutItems() async {
+    items = <ShortcutItem>[];
+    return;
+  }
+
+  @override
+  Future<String?> getLaunchAction() async {
+    getLaunchActionCalled = true;
+    return LAUNCH_ACTION_STRING;
+  }
+
+  @override
+  Future<void> setShortcutItems(List<ShortcutItemMessage?> itemsList) async {
+    await clearShortcutItems();
+    for (final ShortcutItemMessage? element in itemsList) {
+      if (element != null) {
+        items.add(
+            QuickActionsAndroid.shortcutItemMessageToShortcutItem(element));
+      } else {
+        throw Exception('ShortcutItemMessage unexpectedly null');
```

Comment:
Optional: you could skip the if/else logic and just do `element!`, since this exception is basically saying the same thing that a failed force-unwrap exception would say.